client/executor/wasmtime/src/instance_wrapper.rs

```diff
@@ -399,6 +399,7 @@ impl InstanceWrapper {
 	/// See `[memory_as_slice]`. In addition to those requirements, since a mutable reference is
 	/// returned it must be ensured that only one mutable and no shared references to memory exists
 	/// at the same time.
+	#[allow(clippy::mut_from_ref)]
```

Comment:
Can this not be moved down to the `slice::from_raw_parts_mut` call?

Reply:
Nope, it's the sig of the function that it's complaining about (`unsafe fn memory_as_slice_mut(&self) -> &mut [u8]`). The idea that magically you can get a mut ref from only ref inputs. I totally agree it's a suspicious sig, but I'm confident that those unsafes were very carefully reviewed when they were introduced.